airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -0,0 +1,346 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=invalid-name
+"""Interact with Amazon EKS, using the boto3 library."""
+
+import json
+from typing import Dict, List, Optional
+
+from botocore.exceptions import ClientError
+
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+from airflow.utils.json import AirflowJsonEncoder
+
+DEFAULT_RESULTS_PER_PAGE = 100
+DEFAULT_PAGINATION_TOKEN = ''
+
+
+class EKSHook(AwsBaseHook):
```

Comment:
EKSHook or EksHook? Which one do you prefer? Which is recommended by PEP? I asking, because it don't match to base class style.

Reply:
Renamed to EksHook, you are right.  I like EKSHook better because it's an abbreviation but I'll follow conventions.

Reply:
Fixed in 616d249

Reply:
Given it's an acronym keeping it as `EKSHook` is allowed, and just because we made the wrong decision in the past with AwsBaseHook doesn't mean we _have to_ stick by it here :)

Reply:
> Note: When using acronyms in CapWords, capitalize all the letters of the acronym. Thus HTTPServerError is better than HttpServerError.

https://www.python.org/dev/peps/pep-0008/

According to PEP-8, we should use `EKSHook`.

Reply:
ACK.  Will revert this back to EKSHook and have more spine in the future :P

Reply:
Reverted in https://github.com/apache/airflow/commit/667401ff7d1037edf52d17e9e84695edaee52cb5